src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -235,9 +236,34 @@ internal unsafe bool DisconnectExBlocking(SafeSocketHandle socketHandle, int fla
 
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, csep.IPEndPoint);
 
+            if (_socketType == SocketType.Stream && _protocolType == ProtocolType.Tcp)
+            {
+                EnableReuseUnicastPort();
+            }
+
             DoBind(csep.IPEndPoint, csep.SocketAddress);
         }
 
+        private void EnableReuseUnicastPort()
+        {
+            // By enabling SO_REUSE_UNICASTPORT, we defer actual port allocation until the ConnectEx call,
+            // so it can bind to ports from the Windows auto-reuse port range, if configured by an admin.
+            // The socket option is supported on Windows 10+, we are ignoring the SocketError in case setsockopt fails.
```

Comment:
Is there any test we could/should write to help validate this?

Reply:
The problem is, it requires an opt-in registry setting.

Reply:
What about a test that just validates we've configured the socket as best we can? e.g. that we've set SO_REUSE_UNICASTPORT by default?

Reply:
Yeah, that seems worthwhile. @antonfirsov can we add this?

Reply:
Makes sense, will do.